NSTTutorials/UITestCodeGeneration/src/test/java/com/ebay/nst/tutorials/rest/uitestcodegeneration/UITestCodeGenerationTest.java

```diff
@@ -0,0 +1,66 @@
+package com.ebay.nst.tutorials.rest.uitestcodegeneration;
+
+import com.ebay.nst.NSTServiceTestRunner;
+import com.ebay.nst.NSTServiceWrapperProcessor;
+import com.ebay.nst.tutorials.sharedtutorialutilities.rest.CanadaHoliday;
+import com.ebay.runtime.RuntimeConfigManager;
+import com.ebay.softassert.EbaySoftAssert;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public class UITestCodeGenerationTest implements NSTServiceTestRunner {
+
+    private static final String GENERATED_MOCK_PATH = RuntimeConfigManager.getInstance().getIosMocksLocation() + "/UITestCodeGenerationTest_exampleUITestCodeGenerationTest_%d_UITestCodeGenerationWrapper.har";
+
+    @BeforeClass
+    // Remove any prior mocks that were generated when running the test method below.
+    public void removeExistingMocks() {
+        File existingMock1 = new File(String.format(GENERATED_MOCK_PATH, 0));
+        File existingMock2 = new File(String.format(GENERATED_MOCK_PATH, 1));
+        File[] files = new File[]{existingMock1, existingMock2};
+
+        for (File file : files) {
+            if (file.exists()) {
+                file.delete();
+                System.out.printf("Removed existing mock at: %s%n", file.getPath());
+            }
+        }
+    }
+
+    @Test
+    // Example output is placed in the same directory as this file (mockgenerationtutorial).
+    public void exampleUITestCodeGenerationTest() throws Exception {
+        NSTServiceWrapperProcessor serviceProcessor = new NSTServiceWrapperProcessor();
+
+        // Send a GET /api/v1/holidays/{holidayId} request to trigger the ENTRY column being used in the nstToFuiMappingsIos.csv.
+        UITestCodeGenerationWrapper restServiceWrapper = new UITestCodeGenerationWrapper(CanadaHoliday.CANADA_DAY);
+        serviceProcessor.sendRequestAndGetJSONResponse(restServiceWrapper);
+
+        // Send a GET /api/v1/holidays/{holidayId} request to trigger the NAVIGATE column being used in the nstToFuiMappingsIos.csv.
+        restServiceWrapper = new UITestCodeGenerationWrapper(CanadaHoliday.CHRISTMAS_DAY);
+        serviceProcessor.sendRequestAndGetJSONResponse(restServiceWrapper);
+    }
+
+    @AfterClass
+    // This assertion was added to ensure that mock generation output is successful in this tutorial.
+    // This is not a required part of writing an NST test.
+    public void ensureMockIsGeneratedSuccessfully() {
+        File existingMock1 = new File(String.format(GENERATED_MOCK_PATH, 0));
```

Comment:
Not sure about this after class assertion style. Why aren't these part of the test method?

Reply:
We could also define an expected class that has the generated statements in it to compare against the class that is populated at runtime.

Reply:
Done in 3d43f0926bdcdf8d1cddbe7d5466662922cf51bb